dataset/bin.cpp

```diff
@@ -572,8 +582,10 @@ DataArray bin(const Variable &data, const Coords &coords, const Masks &masks,
   const auto masked = hide_masked(data, masks, builder.dims().labels());
   TargetBins<DataArray> target_bins(masked, builder.dims());
   builder.build(*target_bins, bins_view<DataArray>(masked).coords(), coords);
-  return add_metadata(bin<DataArray>(masked, *target_bins, builder), coords,
-                      masks, attrs, builder.edges(), builder.groups(), erase);
+  return add_metadata(bin<DataArray>(drop_grouped_event_coords(masked, groups),
+                                     *target_bins, builder),
+                      coords, masks, attrs, builder.edges(), builder.groups(),
+                      erase);
```

Comment:
Good idea! Is this also needed here? https://github.com/scipp/scipp/blob/8a68869cf45afe912f750dd7a4c20dfe6b0c0e6d/dataset/bin.cpp#L480-L482

Reply:
I think not, see if the comment I just added explains.

Reply:
I think that makes sense